aptos-move/block-executor/src/scheduler.rs

```diff
@@ -186,15 +186,21 @@ impl ValidationStatus {
     }
 }
 
-pub struct Scheduler {
-    /// Number of txns to execute, immutable.
-    num_txns: TxnIndex,
+impl Default for ValidationStatus {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+pub struct Scheduler<P: SchedulerProvider> {
+    provider: Arc<P>,
```

Comment:
provider is not a good name, may be `txn_state_provider` ?

Reply:
So I think it's better to let the provider not hold BlockSTM states, and instead simply provide index operations and other stuff. That's why a generic name is better IMO.